scl/rewrite/cc-mask.conf

```diff
@@ -38,7 +38,14 @@
 # A notable difference compared to the blog post, is that the hash_cc
 # rule is called credit-card-hash and mask_cc is credit-card-mask.
 
-@define balabit.credit-card-regexp "(:4[0-9]{12}(?:[0-9]{3})?|5[1-5][0-9]{14}|6(?:011|5[0-9][0-9])[0-9]{12}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|(?:2131|1800|35\d{3})\d{11})"
+@define balabit.credit-card-regexp.visa "(?:4[0-9]{12}(?:[0-9]{3})?)"
+@define balabit.credit-card-regexp.mastercard "((5[1-5][0-9]{14})|((?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12}))"
+@define balabit.credit-card-regexp.american_express "(3[47][0-9]{13})"
+@define balabit.credit-card-regexp.diners_club "(3(?:0[0-5]|[68][0-9])[0-9]{11})"
+@define balabit.credit-card-regexp.discover "(6(?:011|5[0-9]{2})[0-9]{12})"
+@define balabit.credit-card-regexp.jcb "((?:2131|1800|35[0-9]{3})[0-9]{11})"
+
+@define balabit.credit-card-regexp "(`balabit.credit-card-regexp.visa`|`balabit.credit-card-regexp.mastercard`|`balabit.credit-card-regexp.american_express`|`balabit.credit-card-regexp.diners_club`|`balabit.credit-card-regexp.discover`|`balabit.credit-card-regexp.jcb`)"
```

Comment:
Hmm, this seems like hackish and hard to roll out manually.
I was thinking about using the shiny regex parser, but that won't help here :(

Instead of defines I would suggest to create separate rewrite rule for each credit card type. Something like this:
```
block rewrite credit-card-internal(regex() value() template())
{
   subst("`regex`"
            "`template`"
            value(`value`)
           flags(global, store-matches)
           type(pcre)
    );
};

block rewrite credit-card-visa(value() template())
{
  credit-card-internal(regex("(?:4[0-9]{12}(?:[0-9]{3})?)") value(`value`) template(`template`));
};

//Note: this template argument is just technical to avoid repetition likely won't expose to user or alternative solution is fine
block rewrite credit-card-mask(value("MESSAGE") template("$(substr $1 0 6)******$(substr $1 -4 4)"))
{
    credit-card-visa(value(`value`) template("`template"));
    credit-card-mastercard(value(`value`) template("`template"));
    ...
};
```
It is easier to test different credit card vendors, or extend it with additional one. The downside is that it is now replaces a single pipe with multiple one (but I had my fair share of trouble assuming that regular expression union is fast to evalate, which is not really).